From a2ed367cebdb2f2c66adaad8f3e89bc21d3d82ef Mon Sep 17 00:00:00 2001
From: Enric Balletbo i Serra <eballetbo@iseebcn.com>
Date: Fri, 2 Sep 2011 12:01:45 +0200
Subject: [PATCH 11/44] OMAP3: igep0020: Add board.ei485 early parameter

The board.ei485 parameter allows to use UART1 as RS232 port instead of EI485

  board.ei485=  [yes, no]
    Format: <string>
    no: disable ei485, use UART1 as RS232, supported by igep0020
    default: yes

Signed-off-by: Enric Balletbo i Serra <eballetbo@iseebcn.com>
---
 arch/arm/mach-omap2/board-igep0020.c |   31 +++++++++++++++++++++++++++++++
 1 files changed, 31 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/board-igep0020.c b/arch/arm/mach-omap2/board-igep0020.c
index 2bc38c7..1095dd7 100644
--- a/arch/arm/mach-omap2/board-igep0020.c
+++ b/arch/arm/mach-omap2/board-igep0020.c
@@ -345,6 +345,37 @@ static struct omap_board_mux board_mux[] __initdata = {
 };
 #endif
 
+static int __init ei485_early_param(char *str)
+{
+	char opt[16];
+
+	if (!str)
+		return -EINVAL;
+
+	strncpy(opt, str, 16);
+
+	omap_mux_init_signal("uart1_tx", OMAP_PIN_OUTPUT);
+	omap_mux_init_signal("uart1_rx", OMAP_PIN_INPUT);
+
+	if (!strcmp(opt, "no")) {
+		/*
+		 * To use UART1 as RS232 port instead of RS485 we need configure
+		 * UART1_RTS and UART1_CTS pins in safe mode. To set the RS485
+		 * comment next to lines.
+		 */
+		omap_mux_init_signal("uart1_rts.safe_mode", 0);
+		omap_mux_init_signal("uart1_cts.safe_mode", 0);
+		pr_info("IGEP: board options: ei485=no \n");
+	} else {
+		omap_mux_init_signal("uart1_rts", 0);
+		omap_mux_init_signal("uart1_cts", 0);
+	}
+
+	return 0;
+}
+
+early_param("board.ei485", ei485_early_param);
+
 static void __init igep_init(void)
 {
 	int opt;
-- 
1.7.4.1

